src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -1339,6 +1324,40 @@ export class SettingTextRenderer extends AbstractSettingRenderer implements ITre
 
 		renderValidations(dataElement, template, true);
 	}
+
+	private createInputBox(dataElement: SettingsTreeSettingElement, template: ISettingTextItemTemplate): InputBox {
+		const useMultiline = dataElement.setting.editPresentation === 'multilineText';
+		const inputBoxOptions: IInputOptions = {
+			flexibleHeight: useMultiline,
+			flexibleWidth: false
+		};
+		const inputBox = new InputBox(template.controlElement, this._contextViewService, inputBoxOptions);
+		template.toDispose.add(inputBox);
+		template.toDispose.add(attachInputBoxStyler(inputBox, this._themeService, {
+			inputBackground: settingsTextInputBackground,
+			inputForeground: settingsTextInputForeground,
+			inputBorder: settingsTextInputBorder
+		}));
+		template.toDispose.add(
+			inputBox.onDidChange(e => {
+				if (template.onChange) {
+					template.onChange(e);
+				}
+			})
+		);
+		template.toDispose.add(
+			inputBox.onDidHeightChange(e => {
+				this._onDidChangeSettingHeight.fire({
+					element: dataElement,
+					height: e
```

Comment:
Looks like this is not the height of the setting row, but just the height of the input box. The hard thing here is that normally the setting height (including name, description, padding, etc) is dynamically measured by the list. 

What are the arguments for doing it this way vs using a fixed height multiline input?

Reply:
The main reason for doing it this way would be consistency with the SCM and find widget input boxes.

Reply:
I'm also unsure whether adding an event handler dependent on `dataElement` makes sense, considering how often `dataElement.value` changes. Maybe there is a better way to call into the event upstream, for example, could one pass an array index instead of the element itself?

Reply:
Fixed with https://github.com/microsoft/vscode/pull/127118/commits/d515ee028b306e2d71b205c8c77e7d4e6cc1288e